boards/tenstorrent/tt_blackhole/tt_blackhole_fixed_partitions.dtsi

```diff
@@ -14,101 +20,173 @@ partitions {
 	flash-device-size = <DT_SIZE_M(64)>;
 	flash-block-size = <DT_SIZE_K(4)>;
 
-	cmfwcfg: partition@14000 {
+	/*
+	 * All partitions below here should eventually be mapped to binaries.
+	 * These binaries will not change, and should provide a safe bootloader
+	 * path in the event of an invalid main image.
+	 */
+
+	/*
+	 * Install MCUBoot as the cmfw tag as well as the failover. Otherwise
+	 * the bootrom may fail to load, as it expects a cmfw tag to be
+	 * present.
+	 */
+	boot_partition: partition@14000 {
+		label = "cmfw";
+		reg = <0x14000 DT_SIZE_K(128)>;
+		binary-path = "$BUILD_DIR/mcuboot/zephyr/zephyr.bin";
+	};
+
+	/*
+	 * This partition contains the recovery image. It will be
+	 * loaded by MCUBoot if the primary image fails to validate.
+	 * This binary is not used by the bootrom.
+	 */
+	slot1_partition: partition@34000 {
+		label = "safeimg";
+		reg = <0x34000 DT_SIZE_K(512)>;
+		binary-path = "$BUILD_DIR/recovery/zephyr/zephyr.signed.bin";
+	};
+
+	/*
+	 * This partition defines a binary blob that indicates the
+	 * recovery image is valid and bootable to mcuboot.
+	 */
+	recovery_valid: partition@b3000 {
+		label = "safetail";
+		reg = <0xb3000 DT_SIZE_K(4)>;
+		binary-path = "$BLOBS_DIR/mcuboot_magic_confirmed.bin";
+	};
+
+	/*
+	 * MCUBoot partition- will be loaded by bootrom.
+	 * tagged as fail_over_image in tt-boot-fs,
+	 * used if the primary image fails to validate.
+	 */
+	boot_fallback: partition@b4000 {
+		label = "failover";
+		reg = <0xb4000 DT_SIZE_K(128)>;
+		binary-path = "$BUILD_DIR/mcuboot/zephyr/zephyr.bin";
+	};
+
+	boardcfg: partition@d4000 {
+		label = "boardcfg";
+		reg = <0xd4000 DT_SIZE_K(4)>;
+		binary-path = "$BUILD_DIR/smc/generated_board_cfg/$BOARD_REV/read_only.bin";
+		read-only;
+	};
```

Comment:
I think this should be moved toward the top too, since it's something we likely want to write protect.